apps/analysis/views.py

```diff
@@ -41,16 +43,60 @@ def get_queryset(self):
         return Analysis.objects.filter(project=self.kwargs['project_id']).select_related(
             'project',
             'team_lead',
-        ).prefetch_related('analysispillar_set')
+        )
 
     @action(
         detail=False,
         url_path='summary'
     )
-    def get_summary(self, request, project_id, pk=None, version=None):
+    def get_summary(self, request, project_id, filters=None, pk=None, version=None):
+        from entry.filter_set import get_filtered_entries
+
+        filters = filters or dict()
+        entries_filter_data = filters.get('entries_filter_data', {})
         queryset = self.filter_queryset(self.get_queryset()).filter(project=project_id)
-        serializer = AnalysisSummarySerializer(queryset, many=True, partial=True)
-        return response.Response(serializer.data)
+        queryset = queryset.prefetch_related(
+            models.Prefetch(
+                'analysispillar_set',
+                queryset=AnalysisPillar.objects.defer(
+                    'analysis'
+                ).prefetch_related('analyticalstatement_set')
+            )
+        )
+        total_entries = get_filtered_entries(self.request.user, entries_filter_data).count()
+        total_sources = Lead.objects.filter(
+            project=project_id
+        ).annotate(entries_count=models.Count('entry')).filter(entries_count__gt=0).count()
+        queryset = queryset.annotate(
+            team_lead_name=models.F('team_lead__username'),
+            dragged_entries=models.functions.Coalesce(
+                models.Subquery(
+                    AnalyticalStatement.objects.filter(
+                        analysis_pillar__analysis=models.OuterRef('pk')
+                    ).order_by().values('analysis_pillar__analysis').annotate(count=models.Count('entries', distinct=True))
+                    .values('count')[:1],
+                    output_field=models.IntegerField(),
+                ), 0),
+            discarded_entries=models.functions.Coalesce(
+                models.Subquery(
+                    DiscardedEntry.objects.filter(
+                        analysis_pillar__analysis=models.OuterRef('pk')
+                    ).order_by().values('analysis_pillar__analysis').annotate(count=models.Count('entry', distinct=True))
+                    .values('count')[:1],
+                    output_field=models.IntegerField(),
+                ), 0),
+            total_entries=models.Value(total_entries, output_field=models.IntegerField()),
+            total_sources=models.Value(total_sources, output_field=models.IntegerField())
+        ).annotate(
+            analyzed_entries=models.F('dragged_entries') + models.F('discarded_entries')
```

Comment:
Refactor. Break this into chunks. Move them away from `views.py`.

Probably `models` or `filters`